AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1088,7 +1088,13 @@ - (void)__layout
   // with negative sizes after applying margins, which will cause
   // measureWithSizeRange: on subnodes to assert.
   if (!CGRectEqualToRect(bounds, CGRectZero)) {
+    // Handle placeholder layer creation in case the size of the node changed after the initial placeholder layer
+    // was created
```

Comment:
The first call is:
`-[ASDisplayView willMoveToWindow]` -->  `-[ASDisplayNode __enterHierarchy]` --> `-[ASDisplayNode _setupPlaceholderLayerIfNeeded]`

We setup the _placeholderLayer in this case but we don't assign a placeholderImage as the size is 0.

After layout is done we call `-[ASDisplayNode _completeLayoutCalculation]` and set the `_placeholderImage` in there. The problem is we don't assign it to the `_placeholderLayer` content:

```
if (_placeholderEnabled && [self _displaysAsynchronously] && self.contents == nil) {

  // Zero-sized nodes do not require a placeholder.
  CGSize layoutSize = (_layout ? _layout.size : CGSizeZero);
  if (CGSizeEqualToSize(layoutSize, CGSizeZero)) {
    return;
  }

  if (!_placeholderImage) {
    _placeholderImage = [self placeholderImage];
  }
}

```
The problem in there is we cannot set it in there as it has to be on the main thread.

The fix assign's a new right sized placeholder image after the layout is done that will be set to the already created `_placeholderLayer`'s content